src/rct/printing.py

```diff
@@ -74,11 +74,14 @@ def as_str(self, order):
         s.append("%s:" % _("Order"))
         s.append("\t%s: %s" % (_("Name"), xstr(order.name)))
         s.append("\t%s: %s" % (_("Number"), xstr(order.number)))
+        s.append("\t%s: %s" % (_("Roles"), xstr(order.roles)))
```

Comment:
I think we should not print the header for any attribute which is not included in the source data to ensure compatibility with older versions of Candlepin (both from an entitlement cert content and api perspective).

I think we should list the syspurpose attributes together in the following order after the "Service Type" attribute: Role, Service Level, Usage, Add-ons. As many of the roles (as well as provided products) begin with "Red Hat*" it makes it somewhat difficult to read when the role is listed right after the provided products.

Reply:
I agree that attributes not included in entitlement certificate should not be listed in the output of `subscription-manager list --consumed`.

I implemented the order of attributes according proposal from RFE. I thought that John's proposal was based on discussion with you. I like your proposal more. :-)